apps/api/src/app/events/usecases/digest-filter-steps/digest-filter-steps.usecase.ts

```diff
@@ -46,4 +46,19 @@ export class DigestFilterSteps {
       _templateId: command.templateId,
     };
   }
+  public static getNestedValue<ObjectType>(payload: ObjectType, path: string) {
+    let result = payload;
+    if (!path) return undefined;
+    try {
+      const keys = path.split('.');
+      for (const key of keys) {
+        if (!result) return undefined;
+        result = result[key];
+      }
+    } catch (error) {
+      return undefined;
+    }
+
+    return result;
```

Comment:
I think we could simplify the implementation, but that's why I asked you to add a unit test suite so we can have full confidence in this suggestion because I am not 100% sure:
```suggestion
    if (!path || !payload) {
      return undefined;
    }
    
    try {
      let result;
      const keys = path.split('.');
      for (const key of keys) {
        result = payload[key];
      }
      
      return result;
    } catch (error) {
      return undefined;
    }
```
Let me know what you think or if I missed any nuance of the fix.